bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitTaggedItem.java

```diff
@@ -141,6 +180,28 @@ public boolean isMemberOfAccessoryGroup() {
         return parentGroupItem != null;
     }
 
+    private void parseConfiguration() {
+        if (configuration != null) {
+            Object dimmerModeConfig = configuration.get(DIMMER_MODE);
+            if (dimmerModeConfig instanceof String) {
+                final String dimmerModeConfigStr = (String) dimmerModeConfig;
+                if (dimmerModeConfigStr.equalsIgnoreCase("none")) {
+                    proxyItem.setDimmerMode(HomekitOHItemProxy.DIMMER_MODE_NONE);
+                } else if (dimmerModeConfigStr.equalsIgnoreCase("filterOn")) {
+                    proxyItem.setDimmerMode(HomekitOHItemProxy.DIMMER_MODE_FILTER_ON);
+                } else if (dimmerModeConfigStr.equalsIgnoreCase("filterBrightness100")) {
+                    proxyItem.setDimmerMode(HomekitOHItemProxy.DIMMER_MODE_FILTER_BRIGHTNESS_100);
+                } else if (dimmerModeConfigStr.equalsIgnoreCase("filterOnExceptBrightness100")) {
+                    proxyItem.setDimmerMode(HomekitOHItemProxy.DIMMER_MODE_FILTER_ON_EXCEPT_BRIGHTNESS_100);
+                }
+            }
+            Object delayConfig = configuration.get(DELAY);
+            if (delayConfig instanceof BigDecimal) {
+                proxyItem.setDelay(((BigDecimal) delayConfig).intValue());
+            }
```

Comment:
Might as well cast a wider net on what the config could return.
```suggestion
            if (delayConfig instanceof Number) {
                proxyItem.setDelay(((Number) delayConfig).intValue());
            }
```